src/actions/transactions.test.js

```diff
@@ -128,63 +103,63 @@ describe('actions: transactions', () => {
       });
     });
 
-    afterEach(() => {
-      transactionApiMock.restore();
-      delegateApiMock.restore();
-    });
-
     it('should create an action function', () => {
-      expect(typeof actionFunction).to.be.deep.equal('function');
+      expect(typeof actionFunction).toBe('function');
     });
 
-    it('should dispatch one transactionAddDelegateName action when transaction contains one vote added', () => {
-      const delegateResponse = { username: 'peterpan' };
+    it('should dispatch one transactionAddDelegateName action when transaction contains one vote added', async () => {
       const transactionResponse = {
         asset: {
           votes: [`+${accounts.delegate.publicKey}`],
         },
       };
-      transactionApiMock.returnsPromise().resolves({ data: [transactionResponse] });
-      delegateApiMock.returnsPromise().resolves({ data: [delegateResponse] });
+      const delegateResponse = { username: 'peterpan' };
       const expectedActionPayload = {
         delegate: delegateResponse,
         voteArrayName: 'added',
       };
+      transactionsApi.getSingleTransaction.mockResolvedValue({ data: [transactionResponse] });
+      delegateApi.getDelegate.mockResolvedValue({ data: [delegateResponse] });
 
-      actionFunction(dispatch, getState);
-      expect(dispatch).to.have.been
-        .calledWith({ data: transactionResponse, type: actionTypes.transactionLoaded });
-      expect(dispatch).to.have.been
-        .calledWith({ data: expectedActionPayload, type: actionTypes.transactionAddDelegateName });
+      await actionFunction(dispatch, getState);
+      await setTimeout(() => {});
```

Comment:
This might need some comment to explain why it's needed
```suggestion
      await setTimeout(() => {}); // the timeout ensures that the async code inside `actionFunction ` is called before the next assertion
```